library/platform.c

```diff
@@ -128,6 +128,61 @@ int mbedtls_platform_set_snprintf( int (*snprintf_func)( char * s, size_t n,
 }
 #endif /* MBEDTLS_PLATFORM_SNPRINTF_ALT */
 
+#if defined(_WIN32)
+int mbedtls_platform_win32_vsnprintf( char *s, size_t n, const char *fmt, va_list arg )
+{
+    int ret;
+
+    /* Avoid calling the invalid parameter handler by checking ourselves */
+    if( s == NULL || n == 0 || fmt == NULL )
+        return( -1 );
+
+#if defined(_TRUNCATE)
+    ret = _vsnprintf_s( s, n, _TRUNCATE, fmt, arg );
+#else
+    ret = _vsnprintf( s, n, fmt, arg );
```

Comment:
With or without truncation these functions can return ```-1``` when output buffer is not enough. On the other hand ```vsnprintf``` returns ```n``` or more. If the caller wants to use the return code for determining truncation then this function is not uniform with ```vsnprintf```. Consider making return code uniform. This is up for discussion @sbutcher-arm .

Reply:
That's a fair point, but why on Windows are we calling `_vsnprintf()` and not `vsnprintf()` which is C99 compliant anyway?

To be clear - we need to define that `mbedtls_vsnprintf()` is C99 compliant, and our own implementations MUST conform to that. Unfortunately, this is a blocker on the PR.

Reply:
[Here](https://docs.microsoft.com/en-gb/cpp/c-runtime-library/reference/vsnprintf-vsnprintf-vsnprintf-l-vsnwprintf-vsnwprintf-l) it say:
```
Beginning with the UCRT in Visual Studio 2015 and Windows 10, vsnprintf is no longer identical to _vsnprintf. The vsnprintf function complies with the C99 standard; _vnsprintf is retained for backward compatibility with older Visual Studio code.
```
It might be recent than the [introduction](https://github.com/ARMmbed/mbedtls/commit/6c0c8e0d3de28e711811f4345464c112ad64d390) of ```_vsnprintf``` in mbedtls. May be time for change.